frameworks/elastic/tests/test_searchguard.py

```diff
@@ -0,0 +1,152 @@
+
+
+def _uninstall_services() -> None:
+    sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+
+
+@pytest.fixture
+def sg_internal_users_path(tmp_path) -> str:
+    path = tmp_path / "sg_internal_users.yml"
+    sg_internal_users_config = (
+        # password: admin_password
+        "admin.readonly: true\n"
+        'admin.hash: "$2y$12$Pd3kIQD1WgaKpekPyMkUi.jmBDF3QDmPIEUg37wXCRufZZQOnYYYW"\n'
+        "admin.roles: [ admin ]\n"
+        # password: kibanaserver_password
+        "kibanaserver.readonly: true\n"
+        'kibanaserver.hash: "$2y$12$lrwJYnjrlTGgOcf7Kd6xXOHIbkqFWBO3qgqyLYlEwIuRM3CfgL5fG"\n'
+    )
+
+    path.write_text(sg_internal_users_config)
+    return path.resolve()
+
+
+@pytest.fixture(scope="module")
+def service_account(configure_security: None) -> Iterator[Dict[str, Any]]:
+    service_account_info = transport_encryption.setup_service_account(config.SERVICE_NAME)
+    try:
+
+        yield service_account_info
+    finally:
+        transport_encryption.cleanup_service_account(config.SERVICE_NAME, service_account_info)
+
+
+@pytest.mark.sanity
+def test_searchguard_support_elastic(service_account: Dict[str, Any], sg_internal_users_path: str):
+    try:
+        _uninstall_services()
+        sdk_cmd.run_cli(
+            "security secrets create -f {} {}".format(
+                str(sg_internal_users_path), sg_internal_users_secret_name
+            )
+        )
+
+        sdk_install.install(
+            config.PACKAGE_NAME,
+            service_name=config.SERVICE_NAME,
+            expected_running_tasks=config.DEFAULT_TASK_COUNT,
+            additional_options={
+                "service": {
+                    "name": config.SERVICE_NAME,
+                    "service_account": service_account["name"],
+                    "service_account_secret": service_account["secret"],
+                    "security": {"transport_encryption": {"enabled": True}},
+                },
+                "sgadmin_nodes": {"count": 1},
+                "elasticsearch": {
+                    "searchguard_enabled": True,
+                    "health_user": "admin",
+                    "health_user_password": "admin_password",
+                    "searchguard_internal_users": sg_internal_users_secret_name,
+                },
+            },
+        )
+
+        curl_cmd = "curl -L -i -k -s {}/_searchguard/health".format(
+            "https://" + sdk_hosts.vip_host(config.SERVICE_NAME, "coordinator", 9200)
+        )
+        rc, stdout, stderr = sdk_cmd.service_task_exec(
+            config.SERVICE_NAME, "master-0-node", curl_cmd
+        )
+        assert bool(rc == 0 and stdout and "HTTP/1.1 200" in stdout and '"status":"UP"' in stdout)
+
+    finally:
+        _uninstall_services()
+        sdk_security.delete_secret(sg_internal_users_secret_name)
+
+
+@pytest.mark.sanity
+def test_searchguard_support_kibana(service_account: Dict[str, Any], sg_internal_users_path: str):
```

Comment:
On the first view this test extends logic of previous one. Don't u mind to reuse the code?

Reply:
done! thank you!